wrappers/nodejs/test/test-functional-online.js

```diff
@@ -63,3 +63,193 @@ describe('Points.exportToPly', function() {
+      'RS400_VISUAL_PRESET_MEDIUM_DENSITY',
+      'RS400_VISUAL_PRESET_COUNT',
+    ];
+    const strAttrs = [
+      'rs400_visual_preset_custom',
+      'rs400_visual_preset_default',
+      'rs400_visual_preset_hand',
+      'rs400_visual_preset_high_accuracy',
+      'rs400_visual_preset_high_density',
+      'rs400_visual_preset_medium_density',
+    ];
+    numberAttrs.forEach((attr) => {
+      assert.equal(typeof obj[attr], 'number');
+    });
+    strAttrs.forEach((attr) => {
+      assert.equal(typeof obj[attr], 'string');
+    });
+    for (let i = 0; i < obj.RS400_VISUAL_PRESET_COUNT; i++) {
+      assert.equal(typeof obj.rs400VisualPresetToString(i), 'string');
+    }
+  });
+
+  it('format test', () => {
+    const obj = rs2.format;
+    const numberAttrs = [
+      'FORMAT_ANY',
+      'FORMAT_Z16',
+      'FORMAT_DISPARITY16',
+      'FORMAT_XYZ32F',
+      'FORMAT_YUYV',
+      'FORMAT_RGB8',
+      'FORMAT_BGR8',
+      'FORMAT_RGBA8',
+      'FORMAT_BGRA8',
+      'FORMAT_Y8',
+      'FORMAT_Y16',
+      'FORMAT_RAW10',
+      'FORMAT_RAW16',
+      'FORMAT_RAW8',
+      'FORMAT_UYVY',
+      'FORMAT_MOTION_RAW',
+      'FORMAT_MOTION_XYZ32F',
+      'FORMAT_GPIO_RAW',
+      'FORMAT_6DOF',
+      'FORMAT_DISPARITY32',
+      'FORMAT_COUNT',
+    ];
+    const strAttrs = [
+      'format_any',
+      'format_z16',
+      'format_disparity16',
+      'format_xyz32f',
+      'format_yuyv',
+      'format_rgb8',
+      'format_bgr8',
+      'format_rgba8',
+      'format_bgra8',
+      'format_y8',
+      'format_y16',
+      'format_raw10',
+      'format_raw16',
+      'format_raw8',
+      'format_uyvy',
+      'format_motion_raw',
+      'format_motion_xyz32f',
+      'format_gpio_raw',
+      'format_6dof',
+      'format_disparity32',
+    ];
+    numberAttrs.forEach((attr) => {
+      assert.equal(typeof obj[attr], 'number');
+    });
+    strAttrs.forEach((attr) => {
+      assert.equal(typeof obj[attr], 'string');
+    });
+    for (let i = 0; i < obj.FORMAT_COUNT; i++) {
+      assert.equal(typeof obj.formatToString(i), 'string');
+    }
+  });
+  it('notification_category test', () => {
+    const obj = rs2.notification_category;
+    const numberAttrs = [
+      'NOTIFICATION_CATEGORY_FRAMES_TIMEOUT',
+      'NOTIFICATION_CATEGORY_FRAME_CORRUPTED',
+      'NOTIFICATION_CATEGORY_HARDWARE_ERROR',
+      'NOTIFICATION_CATEGORY_HARDWARE_EVENT',
+      'NOTIFICATION_CATEGORY_UNKNOWN_ERROR',
+    ];
+    const strAttrs = [
+      'notification_category_frames_timeout',
+      'notification_category_frame_corrupted',
+      'notification_category_hardware_error',
+      'notification_category_hardware_event',
+      'notification_category_unknown_error',
+    ];
+    numberAttrs.forEach((attr) => {
+      assert.equal(typeof obj[attr], 'number');
+    });
+    strAttrs.forEach((attr) => {
+      assert.equal(typeof obj[attr], 'string');
+    });
+    for (let i = 0; i < obj.NOTIFICATION_CATEGORY_COUNT; i++) {
```

Comment:
is `obj.NOTIFICATION_CATEGORY_COUNT` an array? If so, prefer to use ES6 for-of

Reply:
It's a number whose values equals the native side NOTIFICATION_CATEGORY_COUNT enum value.